code/modules/clothing/glasses/hud.dm

```diff
@@ -102,6 +102,44 @@
 	hud_trait = TRAIT_SECURITY_HUD
 	glass_colour_type = /datum/client_colour/glass_colour/red
 
+/obj/item/clothing/glasses/hud/security/equipped(mob/living/carbon/human/user, slot)
+	. = ..()
+	if (!.)
+		return FALSE
+	RegisterSignal(user, COMSIG_MOB_MIDDLECLICKON, .proc/create_ping)
```

Comment:
when is the COMISG_MOB_MIDDLECLICKON unregistered, also equipped is called on every single proc that involves moving items in the iventory, so you should also check for slot most likely

Reply:
that is what the parent call is for.